### ISSUE TITLE
Line breaks in multi-word markdown links results in weird output

### ISSUE THREAD

**Issue by digitalsadhu** [prettier/prettier#3203]
**Prettier 1.8.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAqABAbQA4QCYCWArgLbqSxwAeMAugBQAWMM2AzkgPScDmBMjIgCMAdGzCMCQtvDwiCETgDMCUKJ1yFSnGACc4cTiQCGMuLo3GwAa2M84bDfmIkAtBXg0AlAB0oIABoQCGwYBSg2ZFBTGGQlYwAbNjggoV0razgYAGVsK1UeZD0iFJA8CDA4xOSggCs2KgAhdJss7OMSOAAZVTgqpNKIIlZhgCZ+mpA83WTdZBATXWtygHd-IOxdVRgAdQI8AWQADgAGIOSSAiLdEvOChLgARSIIeAnSmGMhPYPGZFGgnpjAQEgUAMIQEgmeZQaB9IJEZIAFS+kSQ8QGAF9MUA)
```sh
--parser markdown
```

**Input:**
```markdown
* [podium context](https://github.schibsted.io/finn/podium/tree/master/packages/podium-context)

```

**Output:**
```markdown
* [podium
  context](https://github.schibsted.io/finn/podium/tree/master/packages/podium-context)

```

**Expected behavior:**
Expect that since the final rendered output is short [podium context] (in this case) that there should not be a line break in the output.

**Comment by azz** [comment]
I'm pretty sure this is to spec. 

cc @ikatyang

**Comment by ikatyang** [comment]
It's breakable there, but it's indeed ugly and looks weird, we should add a special case for it, not sure what the condition should be.

**Comment by ikatyang** [comment]
Oh, the url's head and tail part is also breakable, we should break there.

option 1 (head + tail)

```md
* [podium context](
  https://github.schibsted.io/finn/podium/tree/master/packages/podium-context
  )
```

option 2 (head only)

```md
* [podium context](
  https://github.schibsted.io/finn/podium/tree/master/packages/podium-context)
```

It seems the tail should be considered unbreakable since it looks ugly, right?

**Comment by azz** [comment]
I'd vote for option 2 - provided it works in GFM and commonmark

**Comment by azz** [comment]
Looking at the changes in #3204, maybe we should just never break with `[name](link)` and `![alt](link)` as it looks weird.

**Comment by azz** [comment]
This has been fixed in `1.8.2`.
